connectors/source.py

```diff
@@ -684,6 +689,32 @@ def is_premium():
         """
         return False
 
+    def get_file_extension(self, filename):
+        return get_file_extension(filename)
+
+    def can_file_be_downloaded(self, file_extension, filename, file_size):
```

Comment:
Would it be better to pass the file name (or path?) and calculate extension from it?

I see that now `filename` is used only for debug lines, so method signature feels a bit weird from this perspective.

Reply:
The file extension is needed for other operations (specifically creating a tempfile with the correct suffix), so I extract earlier and pass it on to this function. Also, some connectors get their file extensions from other sources. For example Salesforce responses have a separate field just for the file extension, and it isn't included in the file name.

> I see that now filename is used only for debug lines, so method signature feels a bit weird from this perspective.

Is there something else we can do to fix this?

Reply:
Would it help aesthetically if `filename` was the last arg?

Reply:
It could be that the method returns an object instead of just True/False, e.g.:

```
if file_extension == "":
    return FileDownloadValidationResult(False, SomeEnum.NoExtension) 
```

Names of files and choice of enum/string with reason is up to discussion here, but not relying on `self._logger` here is a nice bonus IMO.

However, it might be a change that can be done only in future when almost the whole download function is in the base source, so my comment can be ignored for now.